src/box/sql/expr.c

```diff
@@ -3192,7 +3192,7 @@ expr_code_int(struct Parse *parse, struct Expr *expr, bool is_neg,
 		size_t len = strlen(z);
 		bool unused;
 		if (sql_atoi64(z, &value, &unused, len) != 0 ||
-		    (is_neg && (uint64_t) value > (uint64_t) INT64_MAX + 1)) {
+		    (is_neg && (uint64_t)value >= (uint64_t)INT64_MAX + 1)) {
```

Comment:
@ImeevMA, the error is not in conversions. The error is lower in `expr.c:3202`

        if (is_neg)
            value = -value;

Here, if the `value` equals to INT64_MIN, `-value` will be INT64_MIN as well (due to the ranges of integer numbers).
Is this correct, that the negation of value will be equal to the same value? I suppose no.

So the patch aimed to prevent the case, when `is_neg` is `true`, and `value` is INT64_MIN, which cannot be negated correctly.

Reply:
Here is actually just a bad code. Here we give value `9223372036854775808` to function `sql_atoi64()` (since `-` was processed before; we got into `expr_code_int()` from `case TK_UMINUS:`) which returns unsigned value `9223372036854775808`. Next we should negate the value, however, since we use `int64_t` instead of `uint64_t`, value is already `-9223372036854775808`.

The proper solution here would be to use `uint64_t value;` instead of `int64_t value;` and use `strtoull()` everywhere after which check the value no more than `INT64_MAX + 1`.